plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1636,7 +1636,7 @@ else if (format == AVRO) {
 
         // date_trunc
         assertThat(query("SELECT * FROM test_day_transform_timestamp WHERE date_trunc('day', d) = TIMESTAMP '2015-05-15 00:00:00'"))
```

Comment:
Let's add here 

```
        assertThat(query("SELECT * FROM test_day_transform_timestamp WHERE date_trunc('day', d) = DATE '2015-05-15'"))
                .isFullyPushedDown();
```

Reply:
i had it like this originally, and I think it may be better to revert to that

Reply:
btw i don't think it's worthwhile testing with both DATE and TIMESTAMP, since actually DATE gets coerced to the TIMESTAMP. The `date_trunc('day', ...)` returns input type, not a date.